src/main/java/org/springframework/data/neo4j/core/Neo4jTemplate.java

```diff
@@ -1191,26 +1198,31 @@ public void setTransactionManager(@Nullable PlatformTransactionManager transacti
 	public <T> ExecutableQuery<T> toExecutableQuery(Class<T> domainType,
 													QueryFragmentsAndParameters queryFragmentsAndParameters) {
 
-		return createExecutableQuery(domainType, null, queryFragmentsAndParameters);
+		return createExecutableQuery(domainType, null, queryFragmentsAndParameters, false);
 	}
 
 
-	private <T> ExecutableQuery<T> createExecutableQuery(Class<T> domainType, @Nullable Class<?> resultType,
- 			QueryFragmentsAndParameters queryFragmentsAndParameters) {
+	private <T> ExecutableQuery<T> createExecutableQuery(
+			Class<T> domainType, @Nullable Class<?> resultType,
+ 			QueryFragmentsAndParameters queryFragmentsAndParameters,
+			boolean readOnly) {
 
 		Supplier<BiFunction<TypeSystem, MapAccessor, ?>> mappingFunction = TemplateSupport
 				.getAndDecorateMappingFunction(neo4jMappingContext, domainType, resultType);
 		PreparedQuery<T> preparedQuery = PreparedQuery.queryFor(domainType)
 				.withQueryFragmentsAndParameters(queryFragmentsAndParameters)
 				.usingMappingFunction(mappingFunction)
 				.build();
-		return toExecutableQuery(preparedQuery);
+		return toQuery(preparedQuery, readOnly);
 	}
 
 	@Override
 	public <T> ExecutableQuery<T> toExecutableQuery(PreparedQuery<T> preparedQuery) {
+        return toQuery(preparedQuery, false);
+	}
 
-		return new DefaultExecutableQuery<>(preparedQuery);
+	private <T> ExecutableQuery<T> toQuery(PreparedQuery<T> preparedQuery, boolean readOnly) {
```

Comment:
```suggestion
	private <T> ExecutableQuery<T> toExecutableQuery(PreparedQuery<T> preparedQuery, boolean readOnly) {
```

minor detail, but I see no reason why this should not have the same name.